# HG changeset patch
# User Torsten <mttl@mailbox.org>
# Date 1520358489 -3600
#      Tue Mar 06 18:48:09 2018 +0100
# Node ID 74229798f604a3829b8d5d1d17baf1dec16711df
# Parent  9adc41f55e3dbb75deaf98fba2951a16cce15e9c
always select suitable icon color for sock widgets depending on title bar color

* octave-dock-widget.cc	(handle_settings): get current color of the title bar
  when not using custom styled title bars and select the button icon colors
  accordingly

diff --git a/libgui/src/octave-dock-widget.cc b/libgui/src/octave-dock-widget.cc
--- a/libgui/src/octave-dock-widget.cc
+++ b/libgui/src/octave-dock-widget.cc
@@ -264,24 +264,33 @@ namespace octave
 
     default_var = QColor (255,255,255);
     m_bg_color = settings->value ("DockWidgets/title_bg_color",
                                   default_var).value<QColor> ();
     default_var = QColor (192,192,192);
     m_bg_color_active = settings->value ("DockWidgets/title_bg_color_active",
                                          default_var).value<QColor> ();
 
+    QColor bcol (m_bg_color);
+    QColor bcola (m_bg_color_active);
+
+    if (! m_custom_style)
+      {
+        bcol = QWidget::palette ().color (m_title_widget->backgroundRole());
+        bcola = bcol;
+      }
+
     int r, g, b;
-    m_bg_color.getRgb (&r, &g, &b);
+    bcol.getRgb (&r, &g, &b);
     if (r+g+b < 400)
       m_icon_color = "-light";
     else
       m_icon_color = "";
 
-    m_bg_color_active.getRgb (&r, &g, &b);
+    bcola.getRgb (&r, &g, &b);
     if (r+g+b < 400)
       m_icon_color_active = "-light";
     else
       m_icon_color_active = "";
 
     notice_settings (settings);  // call individual handler
 
     set_style (false);
